### PR TITLE
feat(@clayui/css): Alerts adds components `.alert-inline`, `alert-autofit-stacked`, `alert-autofit-stacked-sm-down`, `alert-autofit-stacked-xs-down`, `alert-indicator-start`

### LINKED ISSUE #4294: [Alert] Inline Actions
## Proposal

One line
Please read:

- [LEXI-1232](https://issues.liferay.com/browse/LEXI-1232)
- [Inline actions for Alerts ](https://liferay.atlassian.net/l/c/5WgL2bmr)

## Why would adopting this proposal be beneficial?
Most of the alerts used in DXP have two or more lines, mainly because the max-width set constrains them. Providing this option will allow product teams to have wider alerts with actions.

Besides, the Dialect team requested this for their use cases. 

## What are the alternatives to this proposal?
We recommend to not use inline actions when there are more than two lines, in that case to use block actions is a better choice. 

**Responsive Behavior**
Could we change an alert with inline actions to block actions when the screen space is narrowed?

### PR DESCRIPTION
**Pull request by pat270**
fixes #4294

This adds components so we can create Alerts according to the specs mentioned in the issue. These new alerts should still work if tossed into the `alert-notifications-fixed` container.

The markup:

alert-inline-actions

```html
<div class="alert alert-dismissible alert-indicator-start alert-inline alert-success">
  <div class="alert-autofit-row autofit-row">
    <div class="autofit-col autofit-col-expand">
      <div class="autofit-section">
        <span class="alert-indicator"></span>
        <strong class="lead">Success:</strong>
        <span>File uploaded. Would you like to add categories automatically?</span>
      </div>
    </div>
    <div class="autofit-col">
      <div class="autofit-section">
        <div class="btn-group">
          <div class="btn-group-item">
            <button class="btn btn-success btn-sm" type="button">Add</button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button aria-label="Close" class="close" data-dismiss="alert" type="button"></button>
</div>
```

We can use these new utilities to layout the current Toast alert if we wanted.

alert-current

```html
<div class="alert alert-autofit-stacked alert-indicator-start alert-info">
  <div class="alert-autofit-row autofit-row">
    <div class="autofit-col autofit-col-expand">
      <div class="autofit-section">
        <span class="alert-indicator"></span>
        <strong class="lead">With a button</strong>
        This is an alert with a button!
      </div>
    </div>
    <div class="autofit-col">
      <div class="autofit-section">
        <div class="btn-group">
          <div class="btn-group-item">
            <button class="alert-btn btn" type="button">View</button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
```

### CODE REVIEW

**Review by matuzalemsteles — COMMENTED** [review]
@pat270 in general LGTM, just a use case that looks like we need to support looking at the document but I think we already support it https://liferay.atlassian.net/wiki/spaces/ENGLEXICON/pages/1891729698/LEXI-1232+Alert+Inline+actions#Usage, but testing this in this implementation it looks like the alert size is fixed and the text margins don't seem to be respected.

I tried to investigate what is causing this but it seems to be related to the margins added to `.alert-autofit-row` when it is `inline`. But I'm not sure if that would really be it because it would break when there's only one line.

**Comment by pat270** [comment]
@matuzalemsteles I can't seem to visualize what you are saying. Do you have a screenshot or an example I can copy paste?

**Comment by matuzalemsteles** [comment]
of course it is here!

Screen Shot 2021-10-04 at 14 32 27

**Comment by pat270** [comment]
@matuzalemsteles I see now. This is as intended since Lexicon added a rule about maximum two lines for inline alert. I just wanted to keep it simple, but we can add more space between text and top/bottom border. I just need to adjust negative margins on `alert-autofit-row`.

**Comment by matuzalemsteles** [comment]
@pat270 no problem, I just tested it because in the confluence documentation there is this possibility.

**Comment by julien** [comment]
LGTM, but I'd also consider increasing space between the text and the "border"

File: packages/clay-css/src/scss/atlas/variables/_alerts.scss [hunk 1]
```diff
@@ -27,7 +27,7 @@ $alert-close: map-deep-merge(
 	$alert-close
 );
 
-$alert-font-size: 0.875rem !default; // 14px
+$alert-font-size: 1rem !default; // 14px
```
Comment (matuzalemsteles) [inline packages/clay-css/src/scss/atlas/variables/_alerts.scss:30]: @pat270 is this expected? I noticed that the alerts will undergo a visual change, it seems to have gotten bigger, in addition to the font size that has also changed, see this comparison:

|. Before | After |
|--------|-------|
| Screen Shot 2021-10-05 at 18 51 37 | Screen Shot 2021-10-05 at 18 52 24 |

Comment (matuzalemsteles) [reply packages/clay-css/src/scss/atlas/variables/_alerts.scss:30]: Oh ok, no problem.

Comment (pat270) [reply packages/clay-css/src/scss/atlas/variables/_alerts.scss]: This one should be good to go. I changed the wrong variable, should have been `$alert-indicator-font-size`. I also changed the padding so alert height should be the same.

**Review by matuzalemsteles — APPROVED** [review]
LGTM!

**Pull request merged**